onnx/reference/c_ops/c_op_common.h

```diff
@@ -0,0 +1,165 @@
+#if !defined(_CRT_SECURE_NO_WARNINGS)
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace ONNX_NAMESPACE {
+
+namespace py = pybind11;
+
+#define py_array_t py::array_t
+#define py_array_style py::array::c_style | py::array::forcecast
+
+#define is_a_ge_zero_and_a_lt_b(a, b) (static_cast<uint64_t>(a) < static_cast<uint64_t>(b))
+
+#define array2vector(vec, arr, dtype)     \
+  {                                       \
+    if (arr.size() > 0) {                 \
+      auto n = arr.size();                \
+      auto p = (dtype*)arr.data(0);       \
+      vec = std::vector<dtype>(p, p + n); \
+    }                                     \
+  }
+
+#define arrayshape2vector(vec, arr)           \
+  {                                           \
+    if (arr.size() > 0) {                     \
+      vec.resize(arr.ndim());                 \
+      for (size_t i = 0; i < vec.size(); ++i) \
+        vec[i] = (int64_t)arr.shape(i);       \
+    }                                         \
+  }
+
+template <typename T, T b>
+constexpr T roundUpPow2(T a) {
+  return (a + (b - 1)) & (~(b - 1));
+}
+
+template <class NTYPE>
+NTYPE flattened_dimension(const std::vector<NTYPE>& values) {
+  NTYPE r = 1;
+  for (auto it = values.begin(); it != values.end(); ++it)
+    r *= *it;
+  return r;
+}
+
+template <class NTYPE>
+NTYPE flattened_dimension(const std::vector<NTYPE>& values, int64_t first) {
```

Comment:
Nit: https://google.github.io/styleguide/cppguide.html#Function_Names seems to be consistent with the rest in this repo